middleware.go

```diff
@@ -103,6 +105,7 @@ func createMiddleware(mw TykMiddleware) func(http.Handler) http.Handler {
 
 			job.TimingKv("exec_time", time.Since(startTime).Nanoseconds(), meta)
 			job.TimingKv(eventName+".exec_time", time.Since(startTime).Nanoseconds(), meta)
+			mw.Base().UpdateRequestSession(r)
```

Comment:
looks like with current implementation we do this call per each middleware added to chain and it is middleware "scoped" - each middleware processes request, schedules session to be updated via `ctxSetSession ` if needed and then it calls here  `UpdateRequestSession` which does actual  Redis call and calls `ctxDisableSessionUpdate ` - so next middleware will have to schedule it again if session update needed.

when I read 1st idea I though we pass request through all middle-wares and we call update session only once at the very end, which sounds like some special new middleware at the very top of chain (even maybe before cors) which just updates session if it was scheduled for request

Reply:
It is definitely a bug (just fixed it). The main idea behind this line was to run session update for cases when plugin returned `mwStatusRespond`, e.g. responds on its own.

The fix itself was:
```
if errCode != mwStatusRespond {
...
} else {
    mw.Base().UpdateRequestSession(r)
}
```

Reply:
👍